internal/repository/employee/employee_test.go

```
@@ -0,0 +1,422 @@
+		{
+			ID:         "3",
+			FirstName:  "firstname3",
+			LastName:   "lastname3",
+			PositionID: "3",
+		},
+	}
+
+	for _, e := range employees {
+		_ = empRepo.Delete(context.Background(), e.ID)
+	}
+
+	for _, p := range positions {
+		_ = posRepo.Delete(context.Background(), p.ID)
+	}
+}
+
+func TestEmployeeRepository_Create(t *testing.T) {
+	//SetenvEnv(t)
+	config, err := conf.NewConfig()
+	if err != nil {
+		log.Fatalf("Error while getting config: %s", err)
+	}
+
+	db, err := database.ConnectPostgres(config.DB)
+	if err != nil {
+		log.Fatalf("Connection to database failed: %s", err)
+	}
+	defer db.Close()
+
+	positionRepo := position.NewPositionsRepository(db)
+	employeeRepo := NewEmployeesRepository(db, positionRepo)
+
+	InitData(positionRepo, employeeRepo)
+	defer DeleteData(positionRepo, employeeRepo)
+
+	tests := []struct {
+		name        string
+		employee    domain.Employee
+		expectedErr bool
+	}{
+		{
+			name: "OK",
+			employee: domain.Employee{
+
+				ID:         "4",
+				FirstName:  "firstname4",
+				LastName:   "lastname4",
+				PositionID: "1",
+			},
+		},
+		{
+			name: "employee already exists",
+			employee: domain.Employee{
+
+				ID:         "1",
+				FirstName:  "firstname1",
+				LastName:   "lastname1",
+				PositionID: "1",
+			},
+			expectedErr: true,
+		},
+		{
+			name: "position not found",
+			employee: domain.Employee{
+				ID:         "5",
+				FirstName:  "firstname5",
+				LastName:   "lastname5",
+				PositionID: "5",
+			},
+			expectedErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			emp, err := employeeRepo.Create(context.Background(), tt.employee)
+			fmt.Println(err)
+			if (err != nil) != tt.expectedErr {
+				t.Errorf("expected error: %v, got: %s", tt.expectedErr, err)
+			}
+
+			if emp != nil && !reflect.DeepEqual(*emp, tt.employee) {
+				t.Errorf("expected: %v, got: %v", tt.employee, &emp)
+			}
+
+			_ = employeeRepo.Delete(context.Background(), tt.employee.ID)
```

you shouldn't discard error. Either log it if you think it's insignificant or fail the test

---

fixed